v1/topdown/eval.go

```diff
@@ -578,12 +577,12 @@ func (e *eval) evalNot(iter evalIterator) error {
 		child.traceEnter(negation)
 	}
 
-	if err := child.eval(func(*eval) error {
-		if e.traceEnabled {
-			child.traceExit(negation)
-			child.traceRedo(negation)
+	if err := child.eval(func(c *eval) error {
+		if c.traceEnabled {
```

Comment:
Are we sure `c` is the same instance as `child` here? E.g. what if evaluation spawns additional children?